examples/shader/custom_phase_item.rs

```diff
@@ -0,0 +1,422 @@
+        pipeline_cache: SystemParamItem<'w, '_, Self::Param>,
+        pass: &mut TrackedRenderPass<'w>,
+    ) -> RenderCommandResult {
+        // Borrow check workaround.
+        let pipeline_cache = pipeline_cache.into_inner();
+
+        // The render pipeline needs to be in the pipeline cache for this to
+        // succeed.
+        let Some(custom_phase_render_pipeline_id) = maybe_custom_phase_render_pipeline_id else {
+            return RenderCommandResult::Failure;
+        };
+        let Some(custom_phase_render_pipeline) =
+            pipeline_cache.get_render_pipeline(custom_phase_render_pipeline_id.pipeline_id)
+        else {
+            return RenderCommandResult::Failure;
+        };
+
+        pass.set_render_pipeline(custom_phase_render_pipeline);
+        RenderCommandResult::Success
+    }
+}
+
+impl<P> RenderCommand<P> for DrawCustomPhaseItem
+where
+    P: PhaseItem,
+{
+    type Param = SRes<CustomPhaseItemBuffers>;
+
+    type ViewQuery = ();
+
+    type ItemQuery = ();
+
+    fn render<'w>(
+        _: &P,
+        _: ROQueryItem<'w, Self::ViewQuery>,
+        _: Option<ROQueryItem<'w, Self::ItemQuery>>,
+        custom_phase_item_buffers: SystemParamItem<'w, '_, Self::Param>,
+        pass: &mut TrackedRenderPass<'w>,
+    ) -> RenderCommandResult {
+        // Borrow check workaround.
+        let custom_phase_item_buffers = custom_phase_item_buffers.into_inner();
+
+        // Tell the GPU where the vertices are.
+        pass.set_vertex_buffer(
+            0,
+            custom_phase_item_buffers
+                .vertices
+                .buffer()
+                .unwrap()
+                .slice(..),
+        );
+
+        // Tell the GPU where the indices are.
+        pass.set_index_buffer(
+            custom_phase_item_buffers
+                .indices
+                .buffer()
+                .unwrap()
+                .slice(..),
+            0,
+            IndexFormat::Uint32,
+        );
+
+        // Draw one triangle (3 vertices).
+        pass.draw_indexed(0..3, 0, 0..1);
+
+        RenderCommandResult::Success
+    }
+}
+
+impl FromWorld for CustomPhaseItemBuffers {
+    fn from_world(world: &mut World) -> Self {
+        let render_device = world.resource::<RenderDevice>();
+        let render_queue = world.resource::<RenderQueue>();
+
+        // Create the vertex and index buffers.
+        let mut vbo = RawBufferVec::new(BufferUsages::VERTEX);
+        let mut ibo = RawBufferVec::new(BufferUsages::INDEX);
+
+        for vertex in &VERTICES {
+            vbo.push(*vertex);
+        }
+        for index in 0..3 {
+            ibo.push(index);
+        }
+
+        // These two lines are required in order to trigger the upload to GPU.
+        vbo.write_buffer(render_device, render_queue);
+        ibo.write_buffer(render_device, render_queue);
+
+        CustomPhaseItemBuffers {
+            vertices: vbo,
+            indices: ibo,
+        }
+    }
+}
+
+impl Vertex {
```

Comment:
This doesn't really matter so up to you if you want to do anything. I'm personally not a fan of having an impl block so far from the struct. It was a bit surprising to see it so far down.

Arguably, it doesn't matter for the example so it's fine to be down there and not pollute the start of the file which is why I don't mind if it stays there.